cmd/sonobuoy/app/e2e.go

```diff
@@ -0,0 +1,29 @@
+package app
+
+import (
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vmware-tanzu/sonobuoy/pkg/client"
+	"github.com/vmware-tanzu/sonobuoy/pkg/errlog"
+)
+
+type e2eFlags struct {
+	e2efocus          string
+	e2eskip           []string
+	k8sVersion        image.ConformanceImageVersion
+}
+
+func NewCmdE2E() *cobra.Command {
+	f := e2eFlags{}
+	cmd := &cobra.Command{
+		Use:   "e2e",
+		Short: "Generates a list of all tests and tags in that tests",
+		Run:   e2eSonobuoyRun(&f),
```

Comment:
I added the function [e2eSonobuoyRun()](https://github.com/vmware-tanzu/sonobuoy/pull/1533/files#diff-2043e352888b111379ca3bf6fb40fb6e85dde560573ae5ceb2973c64bee34d2cR24) here by observing similar code for func [NewCmdGen()](https://github.com/vmware-tanzu/sonobuoy/blob/main/cmd/sonobuoy/app/gen.go#L195). By observing the behavior of func [genManifest()](https://github.com/vmware-tanzu/sonobuoy/blob/237bd35906f5c4bed1f4de4aa58cc6a6a676d4fd/cmd/sonobuoy/app/gen.go#L207), I think we will be writing the logic for the flags `e2efocus`, `e2eskip`, `k8sVersion` in the func [e2eSonobuoyRun()](https://github.com/vmware-tanzu/sonobuoy/pull/1533/files#diff-2043e352888b111379ca3bf6fb40fb6e85dde560573ae5ceb2973c64bee34d2cR24) which I have added here. 

What will be our focus, working on [e2eSonobuoyRun()](https://github.com/vmware-tanzu/sonobuoy/pull/1533/files#diff-2043e352888b111379ca3bf6fb40fb6e85dde560573ae5ceb2973c64bee34d2cR24) function first or on to adding the subcommands first.
Also, I am not able to figure out what logic to actually add in [e2eSonobuoyRun()](https://github.com/vmware-tanzu/sonobuoy/pull/1533/files#diff-2043e352888b111379ca3bf6fb40fb6e85dde560573ae5ceb2973c64bee34d2cR24) function.